controller/codebase_blackbox_test.go

```diff
@@ -91,10 +142,86 @@ func (s *CodebaseControllerTestSuite) TestDeleteCodebase() {
 				fxt.Spaces[idx].OwnerID = testsupport.TestIdentity.ID
 				return nil
 			}),
-			tf.Codebases(1))
-		// when/then
-		svc, ctrl := s.SecuredControllers(testsupport.TestIdentity)
+			tf.Codebases(1, func(fxt *tf.TestFixture, idx int) error {
+				fxt.Codebases[idx].URL = "git@github.com:bar/foo"
+				return nil
+			}))
+
+		// setup the mock client for Che
+		r, err := recorder.New("../test/data/che/che_delete_codebase_workspaces.ok")
+		require.Nil(t, err)
+		defer r.Stop()
+		m := httpmonitor.NewTransportMonitor(r.Transport)
+		mockCheClient := NewMockCheClient(m, s.Configuration)
+		svc, ctrl := s.SecuredControllers(testsupport.TestIdentity,
+			func(codebaseCtrl *CodebaseController) {
+				codebaseCtrl.NewCheClient = mockCheClient
+			}, func(codebaseCtrl *CodebaseController) {
+				codebaseCtrl.ShowTenant = MockShowTenant()
+			})
+
+		// when
+		test.DeleteCodebaseNoContent(t, svc.Context, svc, ctrl, fxt.Codebases[0].ID)
+		// verify that a `DELETE workspace` request was sent by the Che client
+		err = m.ValidateExchanges(
+			httpmonitor.Exchange{
+				RequestMethod: "GET",
+				RequestURL:    "che-server/workspace?masterUrl=https://tsrv.devshift.net:8443&namespace=foo&repository=git@github.com:bar/foo",
+				StatusCode:    200,
+			},
+			httpmonitor.Exchange{
+				RequestMethod: "DELETE",
+				RequestURL:    "che-server/workspace/string?masterUrl=https://tsrv.devshift.net:8443&namespace=foo",
+				StatusCode:    200,
+			})
+		assert.Nil(t, err)
+
+	})
+
+	s.T().Run("OK with workspace deletion failure", func(t *testing.T) {
+		// given
+		fxt := tf.NewTestFixture(t, s.DB,
+			tf.Spaces(1, func(fxt *tf.TestFixture, idx int) error {
+				fxt.Spaces[idx].OwnerID = testsupport.TestIdentity.ID
+				return nil
+			}),
+			tf.Codebases(1, func(fxt *tf.TestFixture, idx int) error {
+				fxt.Codebases[idx].URL = "git@github.com:bar/foo"
+				return nil
+			}))
+
+		// setup the mock client for Che
+		r, err := recorder.New("../test/data/che/che_delete_codebase_workspaces.failure")
+		require.Nil(t, err)
+		defer r.Stop()
+		r.SetMatcher(func(r *http.Request, i cassette.Request) bool {
+			t.Logf("Matching `%s %s` againt `%s %s`...", r.Method, r.RequestURI, i.Method, i.URL)
+			return cassette.DefaultMatcher(r, i)
+		})
+		m := httpmonitor.NewTransportMonitor(r.Transport)
+		mockCheClient := NewMockCheClient(m, s.Configuration)
+		svc, ctrl := s.SecuredControllers(testsupport.TestIdentity,
+			func(codebaseCtrl *CodebaseController) {
+				codebaseCtrl.NewCheClient = mockCheClient
+			}, func(codebaseCtrl *CodebaseController) {
+				codebaseCtrl.ShowTenant = MockShowTenant()
+			})
+
+		// when
 		test.DeleteCodebaseNoContent(t, svc.Context, svc, ctrl, fxt.Codebases[0].ID)
```

Comment:
Thoughts on e.g. "Ensure Workspace is not deleted when user is not auth" ?

Reply:
well, shouldn't it be covered by the `unauthorized` and `forbidden` subtests already ?

Reply:
That only tests the response. Nothing stopping the impl from actually deleting all the stuff anyway

Reply:
A Delete test is kinda pointless without verifying 404 happens after it's been deleted

Reply:
okay, but here I'm mocking the response using a fake transport.. so what kind of test would you expect ? Testing that Che would actually delete the workspace is not in the scope of this test IMO. 
What I can do, though, is check that the Che client was not called when the user is not authenticated/authorized. Is it what you had in mind ?

Reply:
Yea. Verify Che was not called when it's not expected to be called ;)

Reply:
ok ;)